src/App.tsx

```diff
@@ -15,48 +17,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Name = 'alphabeticaly',
+  Length = 'length',
+}
+
+interface FunctionArguments {
+  sortField : string,
+  reversed: boolean,
+}
+
+function prepareGoods(goods: string[],
+  { sortField, reversed }: FunctionArguments): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Name:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [reversed, setReversed] = useState(false);
```

Comment:
AFAIK, it's better to use Generics via using `useState`

like that:

`const [sortField, setSortField] = useState<string>('');`

Reply:
> AFAIK, it's better to use Generics via using `useState`
> 
> like that:
> 
> `const [sortField, setSortField] = useState<string>('');`

When you set default value to `useState` it understand what type it should, so you don't need to specify with generic in this case